packages/editor/src/components/document-outline/style.scss

```diff
@@ -82,7 +82,8 @@
 	padding: 1px 0;
 }
 
-.editor-document-outline.has-no-headings {
+.document-outline.has-no-headings,
+.document-outline.has-no-main {
```

Comment:
If the warning will be in the `Notice` component, this CSS change is not needed anymore.